edi_voxel/models/res_company.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+
+import pytz
+
+from odoo import fields, models
+
+
+class Company(models.Model):
+    _inherit = "res.company"
+
+    voxel_enabled = fields.Boolean(string="Enable Voxel")
+    voxel_send_mode = fields.Selection(
+        string="Send mode",
+        selection=[
+            ("auto", "On validate"),
+            ("fixed", "At fixed time"),
+            ("delayed", "With delay"),
+        ],
+        default="auto",
+    )
+    voxel_sent_time = fields.Float(string="Sent time")
+    voxel_delay_time = fields.Float(string="Delay time")
+    voxel_login_ids = fields.One2many(
+        comodel_name="voxel.login", inverse_name="company_id", string="Voxel logins",
+    )
+
+    def _get_voxel_report_eta(self):
+        if self.voxel_send_mode == "fixed":
+            tz = self.env.context.get("tz", self.env.user.partner_id.tz)
+            offset = datetime.now(pytz.timezone(tz)).strftime("%z") if tz else "+00"
+            hour_diff = int(offset[:3])
+            hour, minute = divmod(self.voxel_sent_time * 60, 60)
+            hour = int(hour - hour_diff)
+            minute = int(minute)
+            now = datetime.now()
+            if now.hour > hour or (now.hour == hour and now.minute > minute):
+                now += timedelta(days=1)
+            now = now.replace(hour=hour, minute=minute)
+            return now
+        elif self.voxel_send_mode == "delayed":
+            seconds = self.voxel_delay_time * 3600
```

Comment:
Maybe add comment to explain increase delayed time can good idea to debug in future.

Reply:
@victoralmau With this change, do you still think it is necessary to add a comment? https://github.com/OCA/edi/compare/b2698d08fe05d8f3ded2cd95ea2295db760eec2f..e801f4db59c4dc58e46c2262ac32ba6a4ef8eedc#diff-3706652a9a5933dfc74eaa8233200794022dec49183476fd776368f97b681bb0R43-R44

Reply:
Ok, now it's not necessary